packages/wrangler/src/deployment-bundle/apply-middleware.ts

```diff
@@ -54,21 +54,23 @@ export async function applyMiddlewareLoaderFacade(
 		)
 		.join("\n");
 
-	const middlewareFns = middlewareIdentifiers.map(([m]) => `${m}.default`);
+	const middlewareFns = middlewareIdentifiers
+		.map(([m]) => `${m}.default`)
+		.join(",");
 
 	if (entry.format === "modules") {
 		await fs.promises.writeFile(
 			dynamicFacadePath,
 			dedent/*javascript*/ `
 				import worker, * as OTHER_EXPORTS from "${prepareFilePath(entry.file)}";
 				${imports}
-				
-				worker.middleware = [
-					${middlewareFns.join(",")},
-					...(worker.middleware ?? []),
-				].filter(Boolean);
-				
+
 				export * from "${prepareFilePath(entry.file)}";
+
+				export const __INTERNAL_WRANGLER_MIDDLEWARE__ = [
+					...(OTHER_EXPORTS.__INJECT_FOR_TESTING_WRANGLER_MIDDLEWARE__ ?? []),
+					${middlewareFns}
```

Comment:
NIT: I would prefer to put the `join(",")` here rather than in the variable declaration above.